### ISSUE TITLE
DOC: np.random.random documentation refers only to np.random.random_sample

### ISSUE THREAD

**Issue by keithbriggs** [numpy/numpy#7512]
The documentation for [numpy.random.random](https://docs.scipy.org/doc/numpy/reference/generated/numpy.random.random.html) repeatedly refers to numpy.random.random_sample.

**Comment by streeck** [comment]
Hi @keithbriggs, I've submitted a PR for this, could you have a look and see if it's ok?

**Comment by keithbriggs** [comment]
Thanks for this; looks ok to me.


________________________________________

**Comment by ron819** [comment]
If someone going to pick this one up.
Some comments are written in the previous PR:
https://github.com/numpy/numpy/pull/12303

**Comment by bashtage** [comment]
@mattip  These legacy aliases should be dropped in randomgen.  Both `random` and `sample`, and only `random_sample` should remain.

**Comment by mattip** [comment]
That sounds like a deprecation and so must undergo a deprecation cycle

**Comment by rgommers** [comment]
That's not how it sounds to me. "randomgen" is brand new, so nothing needs deprecating there. And the *existing API* should not be touched as part of introducing randomgen.

**Comment by mattip** [comment]
"legacy aliases"

**Comment by rgommers** [comment]
"in randomgen" (= new)

It may be ambiguous. If the suggestion was really to deprecate `np.random.random`, then no - that's definitely a no-go. It's used way too widely, and there's very little to gain from such a deprecation.

**Comment by bashtage** [comment]
There are no plans to depreciate or remove random/RandomState.  In fact, I
checked and these are already gone from RandomGenerator.

**Comment by rgommers** [comment]
This bug report and discussion is about the functions in the `numpy.random` namespace though, not the legacy or new `RandomGenerator`.

To address this problem, we should just fix the `np.random.random` docstring. Just:
```
def random(...):
    """"
    This is an alias of `random_sample`. See the `random_sample` docstring for full documentation.
    """
    random_sample(...)
```
We do this for other aliases as well.

**Comment by bashtage** [comment]
@rgommers This was mostly to link to @mattip 's issue and to be sure that these legacy names don't make it into the new API.

**Comment by rgommers** [comment]
> and to be sure that these legacy names don't make it into the new API.

Indeed, that would be good. New API = new chance to make things clean and consistent.

**Comment by bashtage** [comment]
This is (or will be soon) fixed in #13163 using simple pass-throughs that explain these are aliases.
